### ISSUE TITLE
Snowflake SAMPLE query cannot be parsed when it has placeholders.

### ISSUE THREAD

**Issue by cympfh** [sqlfluff/sqlfluff#4360]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

A query with `SAMPLE` and a placeholder,

```sql
# test.sql
SELECT x FROM t SAMPLE ($p)
```

```toml
# pyproject.toml
[tool.sqlfluff.templater.placeholder]
param_style = "dollar"
p = 20
```

cannot be parsed.

```bash
$ sqlfluff lint --dialect snowflake ./test.sql
== [test.sql] FAIL
L:   1 | P:  17 |  PRS | Line 1, Position 17: Found unparsable section: 'SAMPLE
                       | ($p)'
WARNING: Parsing errors found and dialect is set to 'snowflake'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Expected Behaviour

Placeholders be parsed also in `SAMPLE`

### Observed Behaviour

When it's a number literal instead of `$p`,

```sql
SELECT x FROM t SAMPLE (20)
```

this can be parsed and works.

```bash
   sqlfluff lint --dialect snowflake ./test.sql
All Finished 📜 🎉!
```

### How to reproduce

```bash
$ sqlfluff --version
sqlfluff, version 1.4.5

$ cat test.sql
SELECT x FROM t SAMPLE ($p)

$ cat pyproject.toml
[tool.sqlfluff.templater.placeholder]
param_style = "dollar"
p = 20

$ sqlfluff lint --dialect snowflake ./test.sql
```

### Dialect

snowflake

### Version

1.4.5

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by cympfh** [comment]
This is my fault. it needs

```
[tool.sqlfluff.core]
templater = "placeholder"
```
